contentreview-impl/impl/src/java/org/sakaiproject/turnitin/util/TurnitinLTIUtil.java

```diff
@@ -0,0 +1,182 @@
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.imsglobal.basiclti.BasicLTIUtil;
+import org.sakaiproject.component.api.ServerConfigurationService;
+import org.sakaiproject.lti.api.LTIService;
+
+/**
+ * This is a utility class for wrapping the LTI calls to the TurnItIn Service.
+ * 
+ * @author bgarcia
+ *
+ */
+public class TurnitinLTIUtil {
+	private static final Log log = LogFactory.getLog(TurnitinLTIUtil.class);
+	
+	private static final Log apiTraceLog = LogFactory.getLog("org.sakaiproject.turnitin.util.TurnitinLTIUtil.apicalltrace");
+
+	public static final int BASIC_ASSIGNMENT = 0;
+	public static final int EDIT_ASSIGNNMENT = 1;
+	public static final int INFO_ASSIGNNMENT = 2;
+	public static final int SUBMIT = 3;
+	public static final int RESUBMIT = 4;
+	public static final int INFO_SUBMISSION = 5;
+	
+	private static final String basicAssignmentUrl = "assignment";
+	private static final String editAssignmentUrl = "assignment/edit/";//assignment_id
+	private static final String infoAssignmentUrl = "resource_link_tool/";//assignment_id
+	private static final String submitUrl = "upload/submit/";//assignment_id
+	private static final String resubmitUrl = "upload/resubmit/";//submission_id
+	private static final String infoSubmissionUrl = "outcome_tool_data/";//submission_id
+	
+	private String said = null;
+	private String secret = null;
+	private String globalId = null;
+	private String endpoint = null;
+	private String turnitinSite = null;
+	
+	private LTIService ltiService;
+	public void setLtiService(LTIService ltiService) {
+		this.ltiService = ltiService;
+	}
+	
+	private ServerConfigurationService serverConfigurationService;
+	public void setServerConfigurationService(ServerConfigurationService serverConfigurationService) {
+		this.serverConfigurationService = serverConfigurationService;
+	}
+	
+	public void init() {
+		log.debug("init - TurnitinLTIUtil");
+		if(ltiService == null)
+			log.warn("TurnitinLTIUtil: Could not find LTI service.");
+		
+		said = serverConfigurationService.getString("turnitin.said");
+		secret = serverConfigurationService.getString("turnitin.secretKey");
+		endpoint = serverConfigurationService.getString("turnitin.ltiURL", "https://sandbox.turnitin.com/api/lti/1p0/");		
+		turnitinSite = serverConfigurationService.getString("turnitin.lti.site", "!turnitin");
+	}
+	
+	public boolean makeLTIcall(int type, String urlParam, Map<String, String> ltiProps){
+		try {
+	        
+			HttpClientParams httpParams = new HttpClientParams();
+			httpParams.setConnectionManagerTimeout(60000);
+			HttpClient client = new HttpClient();
+			client.setParams(httpParams);
+			client.getParams().setParameter("http.protocol.content-charset", "UTF-8");
+			
+			//Map<String,String> extra = new HashMap<String,String> ();
+			String extra = "";
+			
+			String defUrl = formUrl(type, urlParam);
+			if(defUrl == null){
+				log.error("Error while getting TII LTI url.");//TODO params
```

Comment:
Much better to actually say what is wrong (type wasn't understood so we couldn't get a URL.